### ISSUE TITLE
Better Error Message: IllegalArgumentException: Unhandled error id: date_out_of_bounds

### ISSUE THREAD

**Issue by hugomg** [grote/Transportr#248]
When I try to search for a route the "Searching for connections..." spinner appears for a couple of seconds then then I get an error message saying 

```
IllegalArgumentException: Unhandled error id: date_out_of_bounds
```

This happens no matter what date I try to use (just now, tomorrow at 10 AM, etc).

I'm using version 1.1.5. I don't know if this bug was present in a previous version because this is the first version I have ever installed.

Is there any extra information I can provide to help figure out what is going on here?

**Comment by grote** [comment]
What provider/network are you using? My guess is that it is Navitia based and that their dataset is out of date. You would need to report that to them since I can't do nothing about it. Try a date some weeks in the past.

**Comment by hugomg** [comment]
Your suspicion was correct. I'm using the Brazil/Rio de Janeiro dataset and according to the Navitia website, that dataset expired 2 weeks ago, on October 12. https://navitia.opendatasoft.com/explore/dataset/br/table/?sort=type_file

It hadn't crossed my mind to test dates in the past and when I tried a date before the 12th the app worked correctly.[1]

That said, perhaps it would be possible to handle the "IllegalArgument" exception and provide a more helpful error message? When I originally saw this error I assumed that the error was my fault, due to me somehow entering an invalid or malformed date. It wasn't clear that the problem was actually an expired dataset.

----------

[1] Unfortunately, the suggested route that I got included some lines that were deactivated more than one year ago. But this is a completely different problem... Now I am investigating if there is something wrong with http://data.rio, which Navitia gets their data from.

**Comment by grote** [comment]
Thanks for looking into the issue!

Let's keep this issue then for improving the error message.

**Comment by patkan** [comment]
I just stumbled across this issue and realized that behavior has changed between master and last release.
The data for Navitia's ParisProvider is not up to date (will report this if it stays until next week)

Search with the following data:
```
Provider Paris
Gare Montparnasse
Porte Maillot
26.11.2016 leaving 10:00
```
The release 1.1.5 will show a toast with the mentioned `IllegalArgumentException`. The current master (f564f0c9d7bff5cc2fbe1ec457fa029311569252) shows a toast with the `error_no_trips_found`-message.

**Comment by grote** [comment]
Yes, there was a change in PTE that changed the error condition. Part of this ticket is to turn all possible `QueryTripsResult`s that are defined in PTE into human readable and translatable error strings to be shown in the Toast.

**Comment by grote** [comment]
This is fixed in 4da6a60e8677103fc2069c952335df6dd273abd6
